### PR TITLE
General refactoring and enhancement for PoPS 1.1 release

### PR DESCRIPTION
**Pull request by malawrim**
Addressing issues #73, #74, and #29. Move general typedefs/enum to utils.hpp and added function to date.hpp.

### CODE REVIEW

**Comment by malawrim** [comment]
I did not fully understand issue #29. I added a new function that returns a string of the current date, but I may have misinterpreted the issue. 

For issue #73, I was not sure about what functions to move. I don't see any that do not seem specific to the file they are in already. Were there specific ones in mind to move?

**Comment by ChrisJones687** [comment]
Could move https://github.com/ncsu-landscape-dynamics/rpops/blob/ec41f5bf0f5ed88b0d74758a782b478102b92f98/src/pops.cpp#L38 to https://github.com/ncsu-landscape-dynamics/rpops/blob/ec41f5bf0f5ed88b0d74758a782b478102b92f98/src/pops.cpp#L98

to the utils.hpp as well. Then when we update rpops once these are part of the next core release we can remove them from rpops.

**Comment by ChrisJones687** [comment]
@petrasovaa is this what you had in mind with #29?

**Comment by petrasovaa** [comment]
> @petrasovaa is this what you had in mind with #29?

Yes, I would just shortened it to 'to_string'. It looks like the implementation would give you "2000-1-1" instead of "2000-01-01"? The latter would be preferable.

File: include/pops/date.hpp [hunk 1]
```diff
@@ -443,6 +444,29 @@ int Date::weeks_from_date(Date start)
...
+    std::string date = std::to_string(year_) + "-";
+    if (month_ < 10) {
+        date = date + "0" + std::to_string(month_) + "-";
+    }
+    else {
+        date = date + std::to_string(month_) + "-";
+    }
+    if (day_ < 10) {
+        date = date + "0" + std::to_string(day_);
+    }
```
Comment (wenzeslaus) [inline include/pops/date.hpp:463]: The current C++ solution for these would be a string stream, but that's actually less readable for non-C++ programmers than this, so just note in the code that this is to be replaced by the C++20 format function in the future.

File: include/pops/date.hpp [hunk 2]
```diff
@@ -443,6 +444,29 @@ int Date::weeks_from_date(Date start)
...
+    else {
+        date = date + std::to_string(day_);
```
Comment (wenzeslaus) [inline include/pops/date.hpp:465]: The usual style would be to use `date += std::to_string(day_);` here which documents the intention (adding to date as opposed to replacing value of date).

**Review by petrasovaa — APPROVED** [review]
Can't comment on template stuff but the rest looks good.

**Review by ChrisJones687 — APPROVED** [review]
Looks good to me. Not sure if the templating from rpops for converting from tuple to vector is the best C++ practice but it works.

**Pull request merged**